sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2392,6 +2399,16 @@ object functions {
   def minute(columnName: String): Column = minute(Column(columnName))
 
   /**
+   * Returns the first date which is later than given date sd and named as dow.
+   * For example, next_day('2015-07-27', 'Sunday') would return 2015-08-02, which is the
+   * first Sunday later than 2015-07-27. The parameter dayOfWeek could be 2-letter, 3-letter,
+   * or full name of the day of the week (e.g. Mo, tue, FRIDAY).
+   * @group datetime_funcs
+   * @since 1.5.0
+   */
+  def next_day(sd: Column, dayOfWeek: Column): Column = NextDay(sd.expr, dayOfWeek.expr)
```

Comment:
i think it is ok to not support that in the data frame api, since it is very unlikely somebody will be using a non literal.